e3sm_diags/driver/arm_diags_driver.py

```diff
@@ -26,168 +30,165 @@
-        if vars_in_file.issuperset(list_of_vars):
-            # All of the variables (list_of_vars) are in data_file.
-            # Return the corresponding dict.
-            return {list_of_vars: vars_to_func_dict[list_of_vars]}  # type: ignore
-
-
-def rmse(predictions, targets):
-    return np.sqrt(((predictions - targets) ** 2).mean())
+# A dictionary that maps diags_set to the appropriate seasons for grouping.
+SEASONS_BY_DIAG: Dict[str, List[ClimoFreq]] = {
+    "diurnal_cycle": ["DJF", "MAM", "JJA", "SON"],
+    "annual_cycle": ["ANNUALCYCLE"],
+    "diurnal_cycle_zt": ["ANNUALCYCLE"],
+    "annual_cycle_aerosol": ["ANNUALCYCLE"],
+}
 
 
-def create_metrics(test, ref):
-    """
-    For this plotset, calculate the mean, std of test (array-like) and ref (array-like), as well as rmse and corr of two datasets and return a dict of that.
+def run_diag(parameter: ARMDiagsParameter) -> ARMDiagsParameter:
+    """Run the specified diagnostic set based on the given ARMDiagsParameter.
+
+    Parameters
+    ----------
+    parameter : ARMDiagsParameter
+        The ARMDiagsParameter object containing the configuration for the diagnostic run.
+
+    Returns
+    -------
+    ARMDiagsParameter
+        The updated ARMDiagsParameter object after running the diagnostic set.
+
+    Raises
+    ------
+    RuntimeError
+        If the specified diags_set is invalid.
     """
-    return {
-        "test_mean": float(test.mean()),
-        "ref_mean": float(ref.mean()),
-        "test_std": float(test.std(ddof=1)),
-        "ref_std": float(ref.std(ddof=1)),
-        "rmse": float(rmse(test, ref)),
-        "corr": float(np.corrcoef(test, ref)[0, 1]),
-    }
+    if parameter.diags_set == "annual_cycle":
+        return _run_diag_annual_cycle(parameter)
+    elif parameter.diags_set == "diurnal_cycle":
+        return _run_diag_diurnal_cycle(parameter)
+    elif parameter.diags_set == "diurnal_cycle_zt":
+        return _run_diag_diurnal_cycle_zt(parameter)
+    elif parameter.diags_set == "pdf_daily":
+        logger.info("'run_diag_pdf_daily' is not yet implemented.")
+    elif parameter.diags_set == "convection_onset":
+        return _run_diag_convection_onset(parameter)
+    elif parameter.diags_set == "aerosol_activation":
+        return _run_diag_aerosol_activation(parameter)
+    if parameter.diags_set == "annual_cycle_aerosol":
+        return _run_diag_annual_cycle_aerosol(parameter)
+    else:
+        raise RuntimeError(f"Invalid diags_set={parameter.diags_set}")
 
 
-def run_diag_diurnal_cycle(parameter: ARMDiagsParameter) -> ARMDiagsParameter:
+def _run_diag_diurnal_cycle(parameter: ARMDiagsParameter) -> ARMDiagsParameter:
     variables = parameter.variables
     regions = parameter.regions
     ref_name = parameter.ref_name
     ref_path = parameter.reference_data_path
 
-    seasons = ["DJF", "MAM", "JJA", "SON"]
+    seasons = SEASONS_BY_DIAG["diurnal_cycle"]
+    test_ds = Dataset(parameter, data_type="test")
```

Comment:
I defined a `SEASONS_BY_DIAG` dictionary since the `seasons` value is static for each set.

Also `test_ds` can defined just once outside of the nested for-loop.